packages/core/src/account/smartContractAccount.ts

```diff
@@ -82,21 +91,28 @@ export type SmartContractAccount<
   getInitCode: () => Promise<Hex>;
   isAccountDeployed: () => Promise<boolean>;
   getFactoryAddress: () => Address;
-  getEntryPoint: () => EntryPointDef<TUO>;
-  getImplementationAddress: () => Promise<"0x0" | Address>;
+  getEntryPoint: () => EntryPointDefRegistry[TEntryPointVersion];
+  getImplementationAddress: () => Promise<NullAddress | Address>;
 };
 //#endregion SmartContractAccount
 
+export interface AccountEntryPointRegistry<Name extends string = string>
```

Comment:
how is this used?

Reply:
looking through the code base this isn't used anywhere actually, we should drop this

Reply:
why was this resolved? This isn't used anywhere so why keep it? This type is already supported by just having `SmartContractAccount<Name, TEntryPointVersion>` and is how we do this in most places

Reply:
this is removed.